src/client/MetaSession.h

```diff
@@ -62,6 +62,13 @@ struct MetaSession {
   MetaSession(mds_rank_t mds_num, ConnectionRef con, const entity_addrvec_t& addrs)
     : mds_num(mds_num), con(con), addrs(addrs) {
   }
+  ~MetaSession() {
+    ceph_assert(caps.empty());
+    ceph_assert(dirty_list.empty());
+    ceph_assert(flushing_caps.empty());
+    ceph_assert(requests.empty());
+    ceph_assert(unsafe_requests.empty());
+  }
```

Comment:
do we know any generic reasons why any/some of these lists would still have entries?

Reply:
I have gone through all the xlist above several time, but couldn't get the root cause. So with this patch we can exactly know which xlist caused this.

Reply:
got it